src/metabase/config.clj

```diff
@@ -41,8 +41,10 @@
    2.  jvm options (ex: -Dmb.db.type -> :mb-db-type)
    3.  hard coded `app-defaults`"
   [k]
-  (let [k (keyword k)]
-    (or (k environ/env) (k app-defaults))))
+  (let [k       (keyword k)
+        env-val (k environ/env)]
+    (or (and (not-empty env-val) env-val)
```

Comment:
why is `and` here needed? And if we expect `env-val` to be string, wouldn't `clojure.string/blank?` be a better predicate?

Reply:
oh good points!

Reply:
Ok, now it clicked why the `and`. Wouldn't an `if` be clearer (I can totally see myself drive-by refactor this to omit the `and` ... and then wasting time figuring out why shit's broken)

Reply:
that's fair, though probably `when` would be preferable

Reply:
for some reason it seemed more natural to use `and` because the enclosing `or` already primed my brain for some boolean operators. maybe it's also because it's a pattern I've seen elsewhere and I've gotten used to it. I don't think I can offer any good reasons to use `and` instead of `when`.

Reply:
@flyingmachine I suspect @sbelak suggested `if` to replace the top-level `or` as well as the `and`:

```clojure
(if (empty? env-val)
  (k app-defaults)
  env-val)
```